_pytest/python.py

```diff
@@ -293,11 +293,11 @@ def pytest_pycollect_makeitem(collector, name, obj):
     elif collector.istestfunction(obj, name):
         # mock seems to store unbound methods (issue473), normalize it
         obj = getattr(obj, "__func__", obj)
-        if not isfunction(obj):
+        if not isfunction(get_real_func(obj)):
             collector.warn(code="C2", message=
                 "cannot collect %r because it is not a function."
                 % name, )
-        if getattr(obj, "__test__", True):
+        elif getattr(obj, "__test__", True):
```

Comment:
I'm a little weary of this change... feels to me it was not accidental that there were two ifs instead of a if/elif chain, but I would like for others to chip in.